chain/indexer/src/streamer/streamer.rs

```diff
@@ -194,11 +197,34 @@ async fn fetch_chunks(
 pub(crate) async fn start(
     view_client: Addr<near_client::ViewClientActor>,
     client: Addr<near_client::ClientActor>,
-    mut blocks_sink: mpsc::Sender<BlockResponse>,
+    indexer_config: IndexerConfig,
+    mut blocks_sink: mpsc::Sender<StreamerMessage>,
 ) {
     info!(target: INDEXER, "Starting Streamer...");
+    let mut indexer_db_path = neard::get_store_path(&indexer_config.home_dir);
+    indexer_db_path.push_str("/indexer");
+
+    // TODO: implement proper error handling
+    let db = DB::open_default(indexer_db_path).unwrap();
     let mut outcomes_to_get = Vec::<types::TransactionOrReceiptId>::new();
-    let mut last_synced_block_height: types::BlockHeight = 0;
+    let mut last_synced_block_height: Option<types::BlockHeight> = match indexer_config.sync_mode {
+        crate::SyncModeEnum::FromInterruption => match db.get(b"last_synced_block_height") {
+            Ok(Some(value)) => Some(String::from_utf8(value).unwrap().parse::<u64>().unwrap()),
+            Ok(None) => None,
+            Err(_) => {
+                db.put(b"last_synced_block_height", b"0").unwrap();
+                None
+            }
```

Comment:
It does not seem to be sane. If we could not read the value, the chances are that we won't be able to write it either, and it also makes quite little sense to put 0 there. Proper error handling or unwrap for the time being seems to be the best option we have here.